src/value_generators/include/value_generators/value_generators.hpp

```diff
@@ -69,6 +77,262 @@ class DocumentGenerator {
  */
 std::unique_ptr<DocumentGenerator> makeDoc(YAML::Node, genny::DefaultRandom&);
 
+
+/**
+ * A variant type for representing different kinds of values that can be generated by an Expression
+ * instance.
+ *
+ * The bsoncxx::types::b_document and bsoncxx::types::b_array cases that are part of
+ * bsoncxx::types::value do not allow for BSON documents or arrays to own their underlying buffer
+ * and can only be constructed from their respective view types.
+ *
+ * Note that we explicitly subclass from std::variant rather than alias it in order to define a
+ * custom std::ostream::operator<< for it.
+ */
+class Value : public std::variant<bool,
```

Comment:
Subclassing from `std::variant` appears to cause compilation to fail with gcc...

https://stackoverflow.com/questions/51309467/using-stdvisit-on-a-class-inheriting-from-stdvariant-libstdc-vs-libc